src/mono/mono/metadata/class.c

```diff
@@ -3582,6 +3582,10 @@ mono_class_is_subclass_of_internal (MonoClass *klass, MonoClass *klassc,
 				    gboolean check_interfaces)
 {
 	MONO_REQ_GC_UNSAFE_MODE;
+
+	if (mono_class_is_pointer(klass) && mono_class_is_pointer(klassc))
```

Comment:
Yeah, that is weird. The unmodified version of that function returns false w/ two pointer types. 

I guess elevate the check here and skip the call if there are two pointer types?

https://github.com/dotnet/runtime/blob/5ea2e7696d7fccab8e8573b16f7b2fed43ad9646/src/mono/mono/metadata/icall.c#L874-L876

Reply:
So this is gonna be a little weird, but I would do soemthing like this:

```c
		if (mono_class_is_pointer (dest_class) && mono_class_is_pointer (src_class)) {
			/* if we're copying between two arrays of pointers, only allow it if both dest_class is assignable from src_class (checked above, and src_class is assignable from dest_class).  This should only be true if both src_class and dest_class have a common cast_class. (for example: int*[] and uint*[] are ok, but void*[] and int*[] are not)). */
			if (!mono_class_is_assignable_from_internal (src_class, dest_class, FALSE))
				return FALSE;
		} else {
			/* It's only safe to copy between arrays if we can ensure the source will always have a subtype of the destination. We bail otherwise. */
			if (!mono_class_is_subclass_of_internal (src_class, dest_class, FALSE))
				return FALSE;
		}